api/client/resources/client.go

```diff
@@ -223,25 +225,18 @@ func (c Client) UploadPendingResource(application string, res charmresource.Reso
 		return "", errors.Trace(err)
 	}
 	pendingID = ids[0]
+	return c.UploadExistingPendingResource(pendingID, application, res, filename, reader)
+}
 
+// UploadExistingPendingResource sends the provided resource blob up to Juju
+// and makes it available.
+func (c Client) UploadExistingPendingResource(pID string, application string, res charmresource.Resource, filename string, reader io.ReadSeeker) (pendingID string, err error) {
```

Comment:
Why have a new method instead of using Upload? Technically it's in the spec, however looking here, it's not needed.

If it is required, why pull in the entire Resources to only use the name?

Reply:
> Why have a new method instead of using Upload? Technically it's in the spec, however looking here, it's not needed.

Because the `c.Upload` is not specifically about resources, therefore doesn't take the pending id as an argument.

Though you're right about the resources name thing, optimizing it now 👍